examples/plot_wlasso.py

```diff
@@ -10,90 +10,96 @@
 alpha_max = np.max(np.abs(X_train.T.dot(y_train))) / X_train.shape[0]
-log_alpha0 = np.log(alpha_max / 10)
 
-n_alphas = 10
-p_alphas = np.geomspace(1, 0.0001, n_alphas)
-alphas = alpha_max * p_alphas
-log_alphas = np.log(alphas)
+n_alphas = 30  # number of iter in the line search, ie 30 evals of the gradient
+alphas = alpha_max * np.geomspace(1, 0.001, n_alphas)
 
-tol = 1e-7
-max_iter = 1e5
+train_split_cv = [train_indices]
+val_split_cv = [val_indices]
 
-##############################################################################
-# Grid-search
-# -----------
 
-# The sklearn solver is indeed very long on the considered problems!
-# estimator = linear_model.Lasso(
-#     fit_intercept=False, max_iter=1000, warm_start=True)
+# vanilla LassoCV
+print("========== Celer's LassoCV started ===============")
+model_cv = LassoCV(verbose=False, fit_intercept=False,
+                   alphas=alphas, tol=1e-7, max_iter=100,
+                   cv=zip(train_split_cv, val_split_cv)).fit(X_train_val,
+                                                             y_train_val)
 
-# celer is way faster !
-# https://github.com/mathurinm/celer
 
-estimator = Lasso(
-    fit_intercept=False, max_iter=1000, warm_start=True)
+# measure mse on test
+mse_cv = mean_squared_error(y_test,  model_cv.predict(X_test))
+print("========== Celer's LassoCV finished ===============")
+print("Vanilla LassoCV: Mean-squared error on test data %f" % mse_cv)
 
 
-##############################################################################
-# Grad-search
-# -----------
-print('sparse-ho started')
+# let's run sparse-ho with vanilla lassoCV initial config
+alpha0 = np.log(model_cv.alpha_) * np.ones(X_train.shape[1])
 
-alpha0 = np.log(alpha_max / 10) * np.ones(n_features)
+###
+# Sparse-ho
+lasso_sho = Lasso(fit_intercept=False, max_iter=1000, warm_start=True)
+model_sho = wLasso(X_train, y_train, estimator=lasso_sho)
+criterion_sho = CV(X_val, y_val, model_sho, X_test=X_test, y_test=y_test)
+algo_sho = ImplicitForward(criterion_sho)
+monitor = Monitor()
+grad_search(
+    algo_sho, alpha0, monitor, n_outer=20, tol=1e-6)
 
-t0 = time.time()
-model = wLasso(X_train, y_train, estimator=estimator)
 
-# Here CV means held out,
-# the "real" crossval (with folds, etc.) is very slow (for the moment) for some
-# unknown reasons
+alphas = np.exp(monitor.log_alphas[-1])
+# X_test /= alphas
+lasso_sho.fit(X_train / alphas, y_train)
+coef_ = lasso_sho.coef_ / alphas
 
-criterion = CV(X_val, y_val, model, X_test=X_test, y_test=y_test)
-algo = ImplicitForward(criterion)
-monitor_grad = Monitor()
-grad_search(
-    algo, alpha0, monitor_grad, n_outer=10, tol=tol)
 
-t_grad_search = time.time() - t0
+# mse on validation set
+mse_sho_val = mean_squared_error(y_val, X_val @ coef_)
+
+# mse on test set, ie unseen data
+mse_sho_test = mean_squared_error(y_test, X_test @ coef_)
 
-print("Time gradient serach:  %f" % t_grad_search)
+print("========== Sparse-ho finished ===============")
```

Comment:
In the end do you think it's reasonable to add a last model, a sparse ho Lasso initialized with celer.LassoCV best alpha, fit it, and check that you get a better test loss with sparse_ho ?

And for plot you could do bar plots of test and val losses for the 3 models

Reply:
>In the end do you think it's reasonable to add a last model, a sparse ho Lasso initialized with celer.LassoCV best alpha, fit it, and check that you get a better test loss with sparse_ho ?
Naively I would say that comparing Lasso CV to a cross val with sparseho is already done in the plot_grad_seach_cv.py example, maybe I should change the name btw

Reply:
>And for plot you could do bar plots of test and val losses for the 3 models
I will add bar plots